exchanges/kraken.py

```diff
@@ -25,15 +30,20 @@ def order_info(self, order_id):
     def trades_history(self, asset_pair, start_time, finish_time):
         pass
 
-    def _public_query(self, query_name, parameters=None):
+    def _public_query(self, query_name, data=None):
         """Queries public data on Kraken"""
 
         # Request public data. This can raise an exception
         # if there is a connection error or if the request times
         # out. In addition, raise an exception if the status is not
         # OK.
         url = '{}public/{}'.format(self.url, query_name)
-        response = requests.get(url, timeout=self.timeout)
+        response = requests.post(url, data=data, timeout=self.timeout)
```

Comment:
Question: Why do we change from get to post? All the public methods are safe, and only used to retrieve data. That is correctly reflected with a "get" request. "Post" should be used only with request that will have some side-effect and need to be secured.